sklearn/impute/_base.py

```diff
@@ -228,7 +228,11 @@ def _validate_input(self, X, in_fit):
                                                         self.strategy))
 
         if self.strategy in ("most_frequent", "constant"):
-            dtype = None
+            if isinstance(X, (list, tuple)) and \
```

Comment:
Since `tuple` is not a typical use case (nor `array-like`), I would remove this data type.